uaclient/tests/test_contract.py

```diff
@@ -235,10 +235,10 @@ def fake_contract_client(cfg):
             fake_client._responses = {
                 API_V1_CONTEXT_MACHINE_TOKEN: ContractAPIError(
                     util.UrlError(
-                        "Server error", code=500, url="http://me", headers={}
+                        "Server error", code=401, url="http://me", headers={}
```

Comment:
Should we be doing more generic "starts with 4xx" here to be flexible?

Reply:
Good point @mitechie I've added 403 as a viable failure case that implies a token error, any other http status codes outside of that seem to imply something other that just invalid token. In these cases the client will still reply with the generic:

```Failed to attach machine. See https://ubuntu.com/advantage```